### PR TITLE
Fixed classification output shape

### PR DESCRIPTION
**Pull request by huntrax11**
The request parameter named `classification_count` could exceed the class count of the model.

https://github.com/triton-inference-server/server/blob/7793fe74eb375307801770251abbc2110b2766cd/src/servers/classification.cc#L46-L64

### CODE REVIEW

**Comment by deadeyegoodwin** [comment]
http_server.cc seems broken in a similar way. The code there could use some other cleanup to consolidate the classification case.

**Comment by huntrax11** [comment]
@deadeyegoodwin Reflected your review.

File: src/servers/grpc_server.cc [hunk 1]
```diff
@@ -2900,7 +2900,7 @@ InferResponseCompleteCommon(
       if (batch_size > 0) {
         output->add_shape(batch_size);
       }
-      output->add_shape(classification_count);
+      output->add_shape(batch1_element_count);
```
Comment (deadeyegoodwin) [inline src/servers/grpc_server.cc:2903]: This needs to be min(classification_count, batch1_element_count)

File: src/servers/http_server.cc [hunk 2]
```diff
@@ -2596,6 +2568,20 @@ HTTPAPIServer::InferRequestClass::FinalizeResponse(
...
+          TRITONSERVER_DataTypeString(TRITONSERVER_TYPE_BYTES);
+      RETURN_IF_ERR(output_json.AddStringRef("datatype", datatype_str));
+
+      TritonJson::Value shape_json(response_json, TritonJson::ValueType::ARRAY);
+      if (batch_size > 0) {
+        RETURN_IF_ERR(shape_json.AppendUInt(batch_size));
+        element_count *= batch_size;
+      }
+      element_count *= batch1_element_count;
+      RETURN_IF_ERR(shape_json.AppendUInt(batch1_element_count));
```
Comment (deadeyegoodwin) [inline src/servers/http_server.cc:2582]: min(info->class_cnt_, batch1_element_count)

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

**Comment by huntrax11** [comment]
@deadeyegoodwin Covered the case you mention.

**Comment by deadeyegoodwin** [comment]
Those changes look ok. Unfortunately our L0_infer CI test is failing with this PR so can't merge it yet. Are you able to run the CI tests? https://docs.nvidia.com/deeplearning/triton-inference-server/master-user-guide/docs/test.html

**Comment by huntrax11** [comment]
@deadeyegoodwin `L0_infer` passes.
```
root@576bb4c846fb:/opt/tritonserver/qa/L0_infer# ./test.sh
=== Running /opt/tritonserver/bin/tritonserver --model-repository=/opt/tritonserver/qa/L0_infer/models --exit-timeout-secs=120
=== Running /opt/tritonserver/bin/tritonserver --model-repository=/opt/tritonserver/qa/L0_infer/models --exit-timeout-secs=120

***
*** Test Passed
***
```

**Review by deadeyegoodwin — APPROVED** [review]

**Pull request merged**